core/src/main/java/org/jdbi/v3/core/ExtensionRegistry.java

```diff
@@ -55,9 +55,10 @@ static ExtensionRegistry copyOf(ExtensionRegistry registry) {
         this.factories = new CopyOnWriteArrayList<>();
     }
 
+    @SuppressWarnings({"unchecked", "rawtypes"}) // working around https://bugs.eclipse.org/bugs/show_bug.cgi?id=499197
     ExtensionRegistry(List<Entry<? extends ExtensionConfig<?>>> factories) {
-        this.factories = factories.stream()
-                .map(Entry::copyOf)
+        this.factories = (List) factories.stream()
+                .map(e -> Entry.copyOf((Entry)e))
```

Comment:
Hm.  It's not convincingly better.  Either way you end up having to have one raw type hack (it ends up being on the `factory.attach(extension, config)` where those types are dependent.  So I guess I'm leaning towards the existing format, which in theory will be fixed in Eclipse 4.7, vs de-generifying which unconditionally requires a raw type escape.